identity/handler.go

```diff
@@ -219,19 +248,67 @@ type AdminCreateIdentityBody struct {
 	// required: true
 	Traits json.RawMessage `json:"traits"`
 
+	// Password for the user. This will only be used if the json schema actually defines the
+	// password as a valid type.
+	//
+	// required: false
+	Password *PasswordCredential `json:"password"`
+
 	// State is the identity's state.
 	//
 	// required: false
 	State State `json:"state"`
+
+	// VerifiableAddresses contains all the addresses that can be verified by the user.
+	//
+	// required: false
+	VerifiableAddresses []VerifiableAddress `json:"verifiable_addresses"`
+}
+
+func (h *Handler) parsePwdCredential(i *Identity, cr AdminCreateIdentityBody, ctx context.Context) error {
+	if cr.Password != nil {
+		// Create the credentials
+		hpw := []byte(cr.Password.Value)
+		if !cr.Password.IsHashed {
+			var err error
+			var hasher hash.Hasher
+			if h.r.Config(ctx).HasherPasswordHashingAlgorithm() == Bcrypt {
```

Comment:
Should PBKDF2 also be added as an option? Looks like support for comparing hashes was merged in #1659

Reply:
Yes! :) Can we maybe use the hash detection we already have from here: https://github.com/ory/kratos/blob/master/hash/hash_comparator.go#L21-L32 ? :)

Reply:
This code is not comparing hashes, is creating them only.  The purpose of this part of the code is to receive a cleartext password and to store it hashed. I haven't found on the existing code any way to  actually create a PBKDF2 hasher to hash a password. There seem to be code only to create a PBKDF2 hasher from an existing hashed password (on https://github.com/ory/kratos/blob/master/hash/hash_comparator.go#L67).

To be able to add support to PBKDF2 here I would need to define a way to configure the PBKDF2 hasher to actually be able to create it. And that's out of the scope for what I wanted to do here.

Something I do not like about my code, though, is that I basically copied the code from 
https://github.com/ory/kratos/blob/master/driver/registry_default.go#L393-L402 because when trying to cast the handler to a RegistryDefault I got a circular dependency.

I've uploaded a new commit to the PR to deal with the circular dependency. To do so:

driver/config imports hash now. hash does not import driver/config. Instead, all the types that hash need are moved to config. After doing that, I found another new circular dependency (schema->config->hash->schema). To remove that one I moved what hash uses from schema (the errors) to its own package. Most of the changes on the commit are from refactoring schema/errors.go to schema/errors/errors.go.

This way the code to construct the hashers is only in one place (in the hash package) and adding new hashers should be simpler. Hopefully.